service/controller/resource/workermigration/security_groups.go

```diff
@@ -0,0 +1,79 @@
+package workermigration
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-11-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	providerv1alpha1 "github.com/giantswarm/apiextensions/v3/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/azure-operator/v5/service/controller/key"
+	"github.com/giantswarm/azure-operator/v5/service/controller/resource/workermigration/internal/azure"
+)
+
+// Security group rules that need destination CIDR update from built-in worker subnet to VNET CIDR.
+var workerSecurityGroupRulesToUpdate = []string{
+	"allowCadvisor",
+	"allowKubelet",
+	"allowNodeExporter",
+	"allowKubeStateMetrics",
+	"sshHostClusterToWorkerSubnetRule",
+}
+
+func (r *Resource) ensureSecurityGroupRulesUpdated(ctx context.Context, cr providerv1alpha1.AzureConfig, azureAPI azure.API) error {
+	var err error
+	var securityGroups azure.SecurityGroups
+	{
+		securityGroups, err = azureAPI.ListNetworkSecurityGroups(ctx, key.ResourceGroupName(cr))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	for _, sg := range securityGroups {
+		if sg.Name != nil && *sg.Name == key.WorkerSecurityGroupName(cr) {
```

Comment:
we have only one security group for all node pools (which is the worker security group), right?

Reply:
Yeah